### PR TITLE
Global username instead of per room

### PR DESCRIPTION
**Pull request by kbariotis**

### CODE REVIEW

File: src/data/localStorage.ts [hunk 1]
```diff
@@ -21,7 +21,14 @@ export function saveUsernameToLocalStorage(id: string, username: string) {
...
   try {
-    const data = localStorage.getItem(`${LOCAL_STORAGE_KEY_COLLAB}-${id}`);
+    // Keep backwards compatibility for those that have already set a username
+    // This will be safe to remove after a couple of days/weeks
+    let data = localStorage.getItem(`${LOCAL_STORAGE_KEY_COLLAB}-${id}`);
+    if (data) {
+      return JSON.parse(data).username;
+    }
+
+    data = localStorage.getItem(`${LOCAL_STORAGE_KEY_COLLAB}`);
```
Comment (lipis) [inline src/data/localStorage.ts:31]:
```suggestion
    data = localStorage.getItem(LOCAL_STORAGE_KEY_COLLAB);
```

File: src/data/localStorage.ts [hunk 2]
```diff
@@ -21,7 +21,14 @@ export function saveUsernameToLocalStorage(id: string, username: string) {
 
 export function restoreUsernameFromLocalStorage(id: string): string | null {
   try {
-    const data = localStorage.getItem(`${LOCAL_STORAGE_KEY_COLLAB}-${id}`);
+    // Keep backwards compatibility for those that have already set a username
+    // This will be safe to remove after a couple of days/weeks
+    let data = localStorage.getItem(`${LOCAL_STORAGE_KEY_COLLAB}-${id}`);
```
Comment (lipis) [inline src/data/localStorage.ts:26]:
```suggestion
    let data = localStorage.getItem(LOCAL_STORAGE_KEY_COLLAB);
```

File: src/data/localStorage.ts [hunk 3]
```diff
@@ -7,10 +7,10 @@ const LOCAL_STORAGE_KEY = "excalidraw";
 const LOCAL_STORAGE_KEY_STATE = "excalidraw-state";
 const LOCAL_STORAGE_KEY_COLLAB = "excalidraw-collab";
 
-export function saveUsernameToLocalStorage(id: string, username: string) {
+export function saveUsernameToLocalStorage(username: string) {
   try {
     localStorage.setItem(
-      `${LOCAL_STORAGE_KEY_COLLAB}-${id}`,
+      `${LOCAL_STORAGE_KEY_COLLAB}`,
```
Comment (lipis) [inline src/data/localStorage.ts:13]:
```suggestion
      LOCAL_STORAGE_KEY_COLLAB,
```

File: src/data/localStorage.ts (hunk 2 above)
Comment (kbariotis) [reply src/data/localStorage.ts:26]: Nop, that's for backward compatibility.

Comment (lipis) [reply src/data/localStorage.ts:26]: The feature was running for a few hours.. on saturday! :)

**Pull request merged**